src/Components/WebAssembly/WebAssembly.Authentication/src/Interop/AuthenticationService.ts

```diff
@@ -52,28 +52,71 @@ export interface AuthenticationResult {
     message?: string;
 }
 
+export interface AuthenticationContext {
+    state?: unknown;
+    interactiveRequest: InteractiveAuthenticationRequest;
+}
+
+export interface InteractiveAuthenticationRequest {
+    scopes?: string[];
+    additionalRequestParameters?: { [key: string]: any };
+};
+
 export interface AuthorizeService {
     getUser(): Promise<unknown>;
     getAccessToken(request?: AccessTokenRequestOptions): Promise<AccessTokenResult>;
-    signIn(state: unknown): Promise<AuthenticationResult>;
+    signIn(context: AuthenticationContext): Promise<AuthenticationResult>;
     completeSignIn(state: unknown): Promise<AuthenticationResult>;
-    signOut(state: unknown): Promise<AuthenticationResult>;
+    signOut(context: AuthenticationContext): Promise<AuthenticationResult>;
     completeSignOut(url: string): Promise<AuthenticationResult>;
 }
 
+export class ManagedLogger {
+    private debug: boolean;
+    private trace: boolean;
+    public constructor(private objectRef: any) {
```

Comment:
I'm not sure about whether JS interop libs are referenced here, but could `objectRef` be typed as `DotNetObjectReference` (or whatever it's called) instead of `any`?

Reply:
No, best to keep it simple. `any` is fine.